From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Fri, 19 Oct 2018 14:42:41 +0200
Subject: [PATCH] don't set saved_entry on grub2-mkconfig

The original plan was for grub2 to rely on the BLS sort criteria to choose
the default entry to boot, to avoid modifying any files when a new kernel
was installed. But that was changed and now 20-grub.install changes the
default, so 10_linux{,bls} shouldn't overwrite this.

Resolves: rhbz#1636466

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 util/grub.d/10_linux.in     | 1 -
 util/grub.d/10_linux_bls.in | 1 -
 2 files changed, 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 01e66e5fc74..b54d2774a7d 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -165,7 +165,6 @@ if [ -s \$prefix/grubenv ]; then
 fi
 EOF
 
-    ${grub_editenv} - set saved_entry=0
     if ! grub2-editenv - list | grep -q kernelopts; then
 	${grub_editenv} - set kernelopts="root=${linux_root_device_thisversion} ro ${args}"
     fi
diff --git a/util/grub.d/10_linux_bls.in b/util/grub.d/10_linux_bls.in
index 1bc97f29898..8745e598d0e 100644
--- a/util/grub.d/10_linux_bls.in
+++ b/util/grub.d/10_linux_bls.in
@@ -212,7 +212,6 @@ linux_entry ()
     populate_header_warn
     populate_menu
 
-    ${grub_editenv} - set saved_entry=0
     if ! grub2-editenv - list | grep -q kernelopts; then
 	${grub_editenv} - set kernelopts="root=${linux_root_device_thisversion} ro ${args}"
     fi
